### PR TITLE
[ROCm] Small functional changes for gptoss

### PR DESCRIPTION
**Pull request by jpvillam-amd**
## Purpose
Small changes to run GPT-OSS on gfx950

## Test Plan
Accuracy test with lm_eval
`lm_eval --model vllm --model_args pretrained=openai/gpt-oss-120b,tensor_parallel_size=8,max_gen_toks=2048 --tasks gsm8k --batch_size auto --num_fewshot 5 --limit 250 --apply_chat_template --output_path ./acc_tmp`

## Test Result
```
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value|   |Stderr|
|-----|------:|----------------|-----:|-----------|---|----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.968|±  |0.0112|
|     |       |strict-match    |     5|exact_match|↑  |0.800|±  |0.0253|
```

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/envs.py [hunk 1]
```diff
@@ -889,6 +890,11 @@ def get_vllm_port() -> Optional[int]:
     env_with_choices("VLLM_ROCM_QUICK_REDUCE_QUANTIZATION", "NONE",
                             ["FP", "INT8", "INT6", "INT4", "NONE"]),
 
+    # Custom layout for weight swizzle mxfp4 moe
+    "VLLM_ROCM_TRITON_HIP_PRESHUFFLE_SCALES":
```
Comment (SageMoore) [inline vllm/envs.py]: It's not obvious to me why you need this environment variable. Just looking at the code, it looks like this is really a GFX950 check? Is that the case? If so can you just check if we are running with GFX950 and remove the environment variable?

**Comment by SageMoore** [comment]
CC: @mgoin

File: vllm/envs.py (hunk 1 above)
Comment (jpvillam-amd) [reply vllm/envs.py]: You are right, we were thinking to allow for non-preshuffled scales but after some debate we don't think its necessary we are going with your suggestion

File: vllm/model_executor/layers/quantization/utils/mxfp4_utils.py [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
 # SPDX-FileCopyrightText: Copyright contributors to the vLLM project
-from typing import Callable, Optional
+from typing import Any, Callable, Optional
 
 import torch
 
+from vllm import envs
 from vllm.logger import init_logger
 from vllm.platforms import current_platform
+from vllm.platforms.rocm import on_gfx950
```
Comment (mgoin) [inline vllm/model_executor/layers/quantization/utils/mxfp4_utils.py]: Can you move this import into the `elif current_platform.is_rocm():` block?

Comment (jpvillam-amd) [reply vllm/model_executor/layers/quantization/utils/mxfp4_utils.py]: Sure thing

**Review by mgoin — APPROVED** [review]
